### PR TITLE
Spark 3.5: Support default values in Parquet reader

### PR DESCRIPTION
**Pull request by rdblue**
This is similar to #11785 but updates the Spark readers.

### CODE REVIEW

**Review by amogh-jahagirdar — APPROVED** [review]

**Comment by manuzhang** [comment]
For the context, is this PR (and previous PRs) resolving https://github.com/apache/iceberg/issues/10761?

File: spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkUtil.java [hunk 1]
```diff
@@ -268,4 +279,59 @@ private static <T> List<T> toJavaList(Seq<T> seq) {
...
   }
+
+  /**
+   * Converts a value to pass into Spark from Iceberg's internal object model.
+   *
+   * @param type an Iceberg type
+   * @param value a value that is an instance of {@link Type.TypeID#javaClass()}
+   * @return the value converted for Spark
+   */
+  public static Object convertConstant(Type type, Object value) {
```
Comment (Fokko) [inline spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkUtil.java:290]: I know that this is a copy of the other one, but I would also expect UUID to be here.

**Review by Fokko — APPROVED** [review]

File: spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkUtil.java (hunk 1 above)
Comment (rdblue) [reply spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkUtil.java:290]: Good point, I'll update it.

Comment (rdblue) [reply spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkUtil.java:290]: The follow up will be here: https://github.com/apache/iceberg/pull/11811

**Pull request merged**